peekingduck/pipeline/nodes/draw/utils/legend.py

```diff
@@ -217,27 +231,34 @@ def _get_item_width(
         text_size = cv2.getTextSize(
             text,
             FONT_HERSHEY_SIMPLEX,
-            SMALL_FONTSCALE,
-            THICK,
+            self.font_size,
+            self.font_thickness,
         )
 
         return text_size[0][0]
 
-    @staticmethod
-    def _get_legend_width(inputs: Dict[str, Any], items: List[str]) -> int:
+    def _get_legend_width(self, inputs: Dict[str, Any]) -> int:
         """Get width of legened box needed to contain all items drawn"""
         max_width = 0
-        for item in items:
+        for item in self.items:
             if item != "zone_count":
-                max_width = max(max_width, Legend._get_item_width(item, inputs[item]))
+                max_width = max(max_width, self._get_item_width(item, inputs[item]))
+            else:
+                max_width = cv2.getTextSize(
+                    ZONE_COUNTS_HEADING,
+                    FONT_HERSHEY_SIMPLEX,
+                    self.font_size,
+                    self.font_thickness,
+                )[0][0]
```

Comment:
When an item before "zone count" is wider, the legend box gets resized to the width of "-ZONE COUNT-" regardless.
![image](https://user-images.githubusercontent.com/56420072/180908495-049de853-4eb8-40b3-b9af-0b4bf7739ebf.png)

Without "zone count", the legend box is resized as expected 
![image](https://user-images.githubusercontent.com/56420072/180908600-4144396d-a19c-40c0-b773-2eb9dc7b6b3e.png)

Also, the current implementation does not consider the case when the "ZONE-<NUM>: <VALUE>" text is wider than the "-ZONE COUNT-" text.
![image](https://user-images.githubusercontent.com/56420072/180908857-bc95f0ca-b5f2-432b-879c-eea1b1d9bf77.png)

Reply:
Implemented some logic to deal with this.